pkg/api/blocks/v1.go

```diff
@@ -55,6 +63,42 @@ func (bapi *BlocksAPI) Register(r *route.Router, tracer opentracing.Tracer, logg
 	instr := api.GetInstr(tracer, logger, ins, logMiddleware, bapi.disableCORS)
 
 	r.Get("/blocks", instr("blocks", bapi.blocks))
+	r.Post("/blocks/mark", instr("blocks_mark", bapi.markBlock))
+}
+
+func (bapi *BlocksAPI) markBlock(r *http.Request) (interface{}, []error, *api.ApiError) {
+	idParam := r.FormValue("id")
+	actionParam := r.FormValue("action")
+	detailParam := r.FormValue("detail")
+
+	if idParam == "" {
+		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: errors.New("ID cannot be empty")}
+	}
+
+	if actionParam == "" {
+		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: errors.New("Action cannot be empty")}
+	}
+
+	id, err := ulid.Parse(idParam)
+	if err != nil {
+		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: errors.Errorf("ULID %q is not valid: %v", idParam, err)}
+	}
+
+	switch actionParam {
+	case "DELETION":
+		err := block.MarkForDeletion(r.Context(), bapi.logger, bapi.bkt, id, detailParam, promauto.With(nil).NewCounter(prometheus.CounterOpts{}))
```

Comment:
We don't need to use a stub counter here. I think it makes sense to use a separate counter to track how many blocks are deleted or marked manually.

Reply:
cc @adzshaf